ingress/controllers/nginx/main.go

```diff
@@ -94,25 +99,36 @@ func main() {
 	}
 
 	if *defaultSvc == "" {
-		glog.Fatalf("Please specify --default-backend")
+		glog.Fatalf("Please specify --default-backend-service")
 	}
 
-	kubeClient, err := unversioned.NewInCluster()
+	var err error
+	if *inCluster {
+		kubeClient, err = unversioned.NewInCluster()
+	} else {
+		config, connErr := clientConfig.ClientConfig()
+		if connErr != nil {
+			glog.Fatalf("error connecting to the client: %v", err)
+		}
+		kubeClient, err = unversioned.New(config)
+	}
 	if err != nil {
 		glog.Fatalf("failed to create client: %v", err)
 	}
 
-	podInfo, err := getPodDetails(kubeClient)
-	if err != nil {
-		glog.Fatalf("unexpected error getting runtime information: %v", err)
+	runtimePodInfo := &podInfo{NodeIP: "127.0.0.1"}
+	if *inCluster {
```

Comment:
Actually this doesn't make sense if it's running in a static pod on the kubernetes master. Will think of something better.